src/System.ComponentModel.TypeConverter/src/System/ComponentModel/BindingList.cs

```diff
@@ -580,7 +511,6 @@ private void Child_PropertyChanged(object sender, PropertyChangedEventArgs e)
 
                     if (pos == -1)
                     {
-                        Debug.Fail("Item is no longer in our list but we are still getting change notifications.");
```

Comment:
You removed this because tests are hitting it?  How about keeping the text as a comment then?